as-ps-diag/as_ps_diag/driver.py

```diff
@@ -39,23 +38,34 @@ def _create_computed_pvs(self):
                              monitor=False)
             self.pvs.append(pv)
             # DiagStatus-Mon
-            pvs = [None, None, None, None, None]
+            pvs = [None, None, None, None, None, None]
             pvs[_PSStatusPV.OPMODE_SEL] = devname + ':OpMode-Sel'
             pvs[_PSStatusPV.OPMODE_STS] = devname + ':OpMode-Sts'
             pvs[_PSStatusPV.CURRT_DIFF] = devname + ':DiagCurrentDiff-Mon'
+            pvs[_PSStatusPV.MAOPMD_SEL] = magname + ':OpMode-Sel'
             pvs[_PSStatusPV.INTLK_SOFT] = devname + ':IntlkSoft-Mon'
             pvs[_PSStatusPV.INTLK_HARD] = devname + ':IntlkHard-Mon'
-            # TODO: Add other interlocks for PS types that hav them
+            # TODO: Add other interlocks for PS types that have them
             pv = _ComputedPV(psname + ':DiagStatus-Mon',
                              _PSStatusPV(),
                              self._queue,
                              pvs,
                              monitor=False)
             self.pvs.append(pv)
 
+    def _get_magname(self, psname):
```

Comment:
would not this function use MASearch.conv_psname_2_psmaname?

Reply:
great call, ana! I will soon commit the change implementing your suggestion!

Reply:
suggestion implemented!